### PR TITLE
Issue #5475 - implement ConfigAccessor.cacheFor

### PR DESCRIPTION
**Pull request by tevans78**
Work in progress, not complete

### CODE REVIEW

File: dev/com.ibm.ws.microprofile.config.1.4/src/com/ibm/ws/microprofile/config14/impl/ConfigAccessorBuilderImpl.java [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+    /**
+     * This method uses a binary count down to append suffixes to a base propertyName.
+     *
+     * e.g.
+     * if there are 4 suffixes then the counter starts at 15 (1111) which would mean that all of the suffixes are applied in order.
+     * The counter then drops to 14 (1110) which would mean all but the last suffix are applied.... and so on until zero where no suffixes are applied
+     *
+     * if propertyName is "BASE" and the suffixes are ONE, TWO, THREE and FOUR then the output is (with counter for reference)
+     *
+     * (1111) BASE.ONE.TWO.THREE.FOUR
```
Comment (tevans78) [inline dev/com.ibm.ws.microprofile.config.1.4/src/com/ibm/ws/microprofile/config14/impl/ConfigAccessorBuilderImpl.java]: @Emily-Jiang FYI, this is my current impl that works out the list of property names with suffixes

**Comment by tevans78** [comment]
#build

File: dev/com.ibm.ws.microprofile.config.1.4/resources/com/ibm/ws/microprofile/config14/resources/Config14.nlsprops [hunk 2]
```diff
@@ -27,6 +27,11 @@
 # Messages used by Config 1.4 Impl -  range is 0300-0399, plus 9996 for the emergency message
 #-----------------------------------------------------------------------------------------------------------------------------
 
+# The property {0} was not found in the configuration.
+no.such.element.CWMCG0300E=CWMCG0300E: The property {0} was not found in the configuration.
+no.such.element.CWMCG0300E.explanation=The requested property did not exist in any of the configured sources.
+no.such.element.CWMCG0300E.useraction=Add the property to one of the existing sources or add a new source that contains the property.
```
Comment (bsbyrd1) [inline dev/com.ibm.ws.microprofile.config.1.4/resources/com/ibm/ws/microprofile/config14/resources/Config14.nlsprops]: Add a comma here:
...sources, or add a new source...

**Comment by tevans78** [comment]
These changes have been incorporated into PR #6083 . Closing this PR.

**Pull request closed without merging**